src/vm/eventpipeconfiguration.cpp

```diff
@@ -409,21 +431,25 @@ void EventPipeConfiguration::DeleteDeferredProviders()
     }
     CONTRACTL_END;
 
-    SListElem<EventPipeProvider*> *pElem = m_pProviderList->GetHead();
-    while(pElem != NULL)
+    // The provider list should not be NULL with reasonable conditions, but check to prevent failure in release builds
+    if (m_pProviderList != NULL)
     {
-        EventPipeProvider *pProvider = pElem->GetValue();
-        if(pProvider->GetDeleteDeferred())
-        {
-            // The act of deleting the provider unregisters it and removes it from the list.
-            delete(pProvider);
-            SListElem<EventPipeProvider*> *pCurElem = pElem;
-            pElem = m_pProviderList->GetNext(pElem);
-            delete(pCurElem);
-        }
-        else
+        SListElem<EventPipeProvider*> *pElem = m_pProviderList->GetHead();
+        while(pElem != NULL)
         {
-            pElem = m_pProviderList->GetNext(pElem);
+            EventPipeProvider *pProvider = pElem->GetValue();
```

Comment:
This looks like it might be overwriting the double delete fix.

Reply:
Thanks for catching that. I've fixed the merge conflict